src/com/maddyhome/idea/vim/action/motion/updown/MotionUpAction.java

```diff
@@ -36,16 +39,26 @@ public MotionUpAction() {
   }
 
   private static class Handler extends MotionEditorActionHandler {
-    public int getOffset(@NotNull Editor editor, DataContext context, int count, int rawCount, Argument argument) {
-      return VimPlugin.getMotion().moveCaretVertical(editor, -count);
+    public Handler() {
+      super(true);
     }
 
-    protected void preMove(@NotNull Editor editor, DataContext context, Command cmd) {
-      col = EditorData.getLastColumn(editor);
+    @Override
+    public int getOffset(@NotNull Editor editor, @NotNull Caret caret, @NotNull DataContext context, int count,
```

Comment:
There is a similar bug in the up action as it is with the down action (see the example).